src/mono/wasm/runtime/jiterpreter-trampolines.ts

```diff
@@ -0,0 +1,415 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { mono_assert, MonoMethod, MonoType } from "./types";
+import { NativePointer } from "./types/emscripten";
+import { Module } from "./imports";
+import {
+    getU32, _zero_region
+} from "./memory";
+import { WasmOpcode } from "./jiterpreter-opcodes";
+import cwraps from "./cwraps";
+import {
+    WasmValtype, WasmBuilder, add_trace_function_pointer,
+    _now, elapsedTimes, counters, getRawCwrap, importDef
+} from "./jiterpreter-support";
+
+// Controls miscellaneous diagnostic output.
+const trace = 0;
+const
+    // Dumps all compiled wrappers
+    dumpWrappers = false;
+
+const instrumentedMethodNames : Array<string> = [
+    // "GetMemberAndWriteJson"
+];
+
+/*
+typedef struct {
```

Comment:
Is this the conceptual stack frame passed to wasm trace ? 
Short explainer here would help a lot. Also about the rest of the file.